### ISSUE TITLE
Enum case casing sniff

### ISSUE THREAD

**Issue by dereuromark** [slevomat/coding-standard#1649]
I saw the following sniffs around enums
```xml
    <rule ref="SlevomatCodingStandard.Classes.BackedEnumTypeSpacing"/>
    <rule ref="SlevomatCodingStandard.Classes.EnumCaseSpacing"/>
```

But I think it would also be a good idea to have the recommended casing for enum cases to be added
Refs https://www.php.net/manual/en/language.enumerations.backed.php

See also
- https://github.com/php-fig/per-coding-style/blob/master/spec.md#9-enumerations
- https://symfony.com/doc/current/contributing/code/standards.html

```php
enum Suit: string
{
    case Hearts = 'H';
    case SOME_SPADES = 'S';
}
```
should be fixed to
```php
enum Suit: string
{
    case Hearts = 'H';
    case SomeSpades = 'S';
}
```
